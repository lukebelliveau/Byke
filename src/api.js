// @flow
export const url = 'http://byke-graphql.herokuapp.com/graphql?query=';

const googleMapsAPIKey = 'AIzaSyAQhawlRtPkYdpC4VHgaB1n3yLOVOEZbYE';

const getAllStations = (stationParams: string) =>
  fetch(`${url}{allStations${stationParams}}`)
    .then(response => {
      return response.json().then(res => {
        return res.data.allStations;
      });
    })
    .catch(e => console.log(e));

const searchPlaces = (searchQuery: string, latitude, longitude) =>
  fetch(
    `https://maps.googleapis.com/maps/api/place/nearbysearch/json?key=${googleMapsAPIKey}&location=${latitude},${longitude}&radius=10000&keyword=${searchQuery}`
  )
    .then(response => {
      response.json().then(json => json.results);
    })
    .catch(e => console.log(e));

export default {
  getAllStations,
  searchPlaces,
};
