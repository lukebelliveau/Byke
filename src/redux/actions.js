export const types = {
  LOADING_FINISHED: 'LOADING_FINISHED',
  LOADING_STARTED: 'LOADING_STARTED',
  STATIONS_FETCHED: 'STATIONS_FETCHED',
};

const loadingStarted = () => ({
  type: types.LOADING_STARTED,
});

const loadingFinished = () => ({
  type: types.LOADING_FINISHED,
});

const stationsFetched = stations => ({
  type: types.STATIONS_FETCHED,
  payload: stations,
});

export default {
  loadingStarted,
  loadingFinished,
  stationsFetched,
};
