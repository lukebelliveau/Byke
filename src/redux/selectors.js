// @flow
import { State } from '../Types';
const getCoordinates = (state: State) => ({
  latitude: state.region.latitude,
  longitude: state.region.longitude,
});

export default {
  getCoordinates,
};
