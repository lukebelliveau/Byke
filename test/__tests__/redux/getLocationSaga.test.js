import SagaTester from 'redux-saga-tester';
import { select } from 'redux-saga';

import getLocationSaga from '../../../src/redux/getLocationSaga';
import reducer from '../../../src/redux/reducers';
import actions, { types } from '../../../src/redux/actions';
import places from '../../testLocations';
import api from '../../../src/api';
import selectors from '../../../src/redux/selectors';

let state = reducer();

let sagaTester;

// beforeEach(() => {
//   sagaTester = new SagaTester({ state });
//   sagaTester.start(getLocationSaga);
// });
//
// api.searchPlaces = jest.fn(() => ({ results: places.results }));
// selectors.getCoordinates = jest.fn(() => ({ latitude: 50, longitude: 50 }));

it('sets currentLocation', () => {
  // sagaTester.dispatch(actions.searchPlaces('local place'));
  //
  // await sagaTester.waitFor(types.PLACES_FETCHED);
  //
  // expect(sagaTester.getLatestCalledAction()).toEqual(
  //   actions.placesFetched(places.results)
  // );
});

// const mockGeolocation = {
//   getCurrentPosition: jest.fn(callback => {
//     callback({
//       coords: { latitude: 50, longitude: 50 },
//     });
//   }),
//   watchPosition: jest.fn(callback => {
//     callback({
//       coords: { latitude: 50, longitude: 50 },
//     });
//   }),
// };
//
// global.navigator.geolocation = mockGeolocation;
