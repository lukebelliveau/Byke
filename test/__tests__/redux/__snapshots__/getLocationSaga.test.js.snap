// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays stations from state 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <MapView
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <Marker
        coordinate={
          Object {
            "latitude": 50,
            "longitude": 50,
          }
        }
      />
      <Marker
        coordinate={
          Object {
            "availableBikes": 1,
            "latitude": 50,
            "longitude": 60,
            "stationName": "station1",
          }
        }
        description="1 bikes available"
        onPress={[Function]}
        title="station1"
      >
        <View
          style={
            Object {
              "backgroundColor": "red",
              "borderRadius": 10,
              "height": 15,
              "width": 15,
            }
          }
        />
      </Marker>
    </View>
  </MapView>
</View>
`;
